mocker.resetall() doesn't work anymore when using mocker.spy()

Using the same example from the README about `.spy()` but adding a `mocker.resetall()`
```
$ cat test_foo.py 
def test_spy_method(mocker):
    class Foo(object):
        def bar(self, v):
            return v * 2

    foo = Foo()
    spy = mocker.spy(foo, 'bar')
    assert foo.bar(21) == 42

    spy.assert_called_once_with(21)
    assert spy.spy_return == 42

    mocker.resetall()

$ pytest test_foo.py 
============================================= test session starts ==============================================
platform linux -- Python 3.8.0, pytest-6.2.3, py-1.10.0, pluggy-0.13.1
rootdir: /home/blaxter
plugins: mock-3.6.0
collected 1 item                                                                                               

test_foo.py F                                                                                            [100%]

=================================================== FAILURES ===================================================
_______________________________________________ test_spy_method ________________________________________________

mocker = <pytest_mock.plugin.MockerFixture object at 0x7f376c49ddc0>

    def test_spy_method(mocker):
        class Foo(object):
            def bar(self, v):
                return v * 2
    
        foo = Foo()
        spy = mocker.spy(foo, 'bar')
        assert foo.bar(21) == 42
    
        spy.assert_called_once_with(21)
        assert spy.spy_return == 42
    
>       mocker.resetall()

test_foo.py:13: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pytest_mock.plugin.MockerFixture object at 0x7f376c49ddc0>

    def resetall(
        self, *, return_value: bool = False, side_effect: bool = False
    ) -> None:
        """
        Call reset_mock() on all patchers started by this fixture.
    
        :param bool return_value: Reset the return_value of mocks.
        :param bool side_effect: Reset the side_effect of mocks.
        """
        for m in self._mocks:
>           m.reset_mock(return_value=return_value, side_effect=side_effect)
E           TypeError: reset_mock() got an unexpected keyword argument 'return_value'

.virtualenvs/test/lib/python3.8/site-packages/pytest_mock/plugin.py:73: TypeError
=========================================== short test summary info ============================================
FAILED test_foo.py::test_spy_method - TypeError: reset_mock() got an unexpected keyword argument 'return_value'
============================================== 1 failed in 0.02s ===============================================
 ```

Same for me, but I'm not using spy. 


```
>       mocker.resetall()

tests/test_cf/test_unit/test_functions.py:531: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pytest_mock.plugin.MockerFixture object at 0x7f42a97a4160>

    def resetall(
        self, *, return_value: bool = False, side_effect: bool = False
    ) -> None:
        """
        Call reset_mock() on all patchers started by this fixture.
    
        :param bool return_value: Reset the return_value of mocks.
        :param bool side_effect: Reset the side_effect of mocks.
        """
        for m in self._mocks:
>           m.reset_mock(return_value=return_value, side_effect=side_effect)
E           TypeError: reset_mock() got an unexpected keyword argument 'return_value'

../../../.cache/pypoetry/virtualenvs/cloud-radar-oC20LMki-py3.9/lib/python3.9/site-packages/pytest_mock/plugin.py:73: TypeError
```

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -1,3 +1,12 @@
+3.6.1 (UNRELEASED)
+------------------
+
+* Fix ``mocker.resetall()`` when using ``mocker.spy()`` (`#237`_). Thanks `@blaxter`_ for the report and `@shadycuz`_ for the PR.
+
+.. _@blaxter: https://github.com/blaxter
+.. _@shadycuz: https://github.com/shadycuz
+.. _#237: https://github.com/pytest-dev/pytest-mock/issues/237
+
 3.6.0 (2021-04-24)
 ------------------
 

--- a/src/pytest_mock/plugin.py
+++ b/src/pytest_mock/plugin.py
@@ -16,6 +16,7 @@ from typing import Mapping
 from typing import Optional
 from typing import overload
 from typing import Tuple
+from typing import Type
 from typing import TypeVar
 from typing import Union
 
@@ -69,13 +70,18 @@ class MockerFixture:
         :param bool return_value: Reset the return_value of mocks.
         :param bool side_effect: Reset the side_effect of mocks.
         """
+        supports_reset_mock_with_args: Tuple[Type[Any], ...]
+        if hasattr(self, "AsyncMock"):
+            supports_reset_mock_with_args = (self.Mock, self.AsyncMock)
+        else:
+            supports_reset_mock_with_args = (self.Mock,)
+
         for m in self._mocks:
-
-            if isinstance(m, self.Mock):
+            # See issue #237.
+            if isinstance(m, supports_reset_mock_with_args):
                 m.reset_mock(return_value=return_value, side_effect=side_effect)
-                continue
-
-            m.reset_mock()
+            else:
+                m.reset_mock()
 
     def stopall(self) -> None:
         """

--- a/src/pytest_mock/plugin.py
+++ b/src/pytest_mock/plugin.py
@@ -70,7 +70,12 @@ class MockerFixture:
         :param bool side_effect: Reset the side_effect of mocks.
         """
         for m in self._mocks:
-            m.reset_mock(return_value=return_value, side_effect=side_effect)
+
+            if isinstance(m, self.Mock):
+                m.reset_mock(return_value=return_value, side_effect=side_effect)
+                continue
+
+            m.reset_mock()
 
     def stopall(self) -> None:
         """

--- a/tests/test_pytest_mock.py
+++ b/tests/test_pytest_mock.py
@@ -310,7 +310,7 @@ def test_spy_reset(mocker: MockerFixture) -> None:
     assert spy.spy_return == 30
     assert spy.spy_exception is None
 
-    # Testing spy can still be reset.
+    # Testing spy can still be reset (#237).
     mocker.resetall()
 
     with pytest.raises(ValueError):

--- a/tests/test_pytest_mock.py
+++ b/tests/test_pytest_mock.py
@@ -310,6 +310,9 @@ def test_spy_reset(mocker: MockerFixture) -> None:
     assert spy.spy_return == 30
     assert spy.spy_exception is None
 
+    # Testing spy can still be reset.
+    mocker.resetall()
+
     with pytest.raises(ValueError):
         Foo().bar(0)
     assert spy.spy_return is None

